### DISCUSSION TITLE
Source or sink term in stokes solution

### DISCUSSION THREAD

**Discussion by kunal0kumar0sinha** [PMEAL/OpenPNM#2783] [Development]
Is there a provision to get a source or sink in the Stokes flow algorithm?

**Comment by jgostick** [comment]
Yes, but source or sink of what?  If you want to consume the flowing fluid then things become a bit questionable physically.  If you want to consume a solute in a fluid then you want the [AdvectionDiffusion](https://openpnm.org/examples/applications/dispersion_coefficient.html) algorithm.

**Reply by kunal0kumar0sinha** [reply]
Yes, I require a sink for the fluid itself, as the porous solid absorbs fluid, while the fluid flows through it.

**Reply by jgostick** [reply]
Then you can start by taking the FickianDiffusion example and applying it to your case.  You will have to write your own 'sink' term model though.  All of the models in OpenPNM were written with reactions in mind. Actually, this makes me think "what is your sink term dependent on?".  To use the openpnm reaction mechinery it needs to be dependent on pressure, which is the variable being solved for in StokesFlow.
